components/runtime/src/ts/discord/permissions.ts

```diff
@@ -0,0 +1,206 @@
+    static AddReactions             = 1n << 6n;
+    static ViewAuditLog             = 1n << 7n;
+    static PrioritySpeaker          = 1n << 8n;
+    static Stream                   = 1n << 9n;
+    static ViewChannel              = 1n << 10n;
+    static SendMessages             = 1n << 11n;
+    static SendTtsMessages          = 1n << 12n;
+    static ManageMessages           = 1n << 13n;
+    static EmbedLinks               = 1n << 14n;
+    static AttachFiles              = 1n << 15n;
+    static ReadMessageHistory       = 1n << 16n;
+    static MentionEveryone          = 1n << 17n;
+    static UseExternalEmojis        = 1n << 18n;
+    static ViewGuildInsights        = 1n << 19n;
+    static Connect                  = 1n << 20n;
+    static Speak                    = 1n << 21n;
+    static MuteMembers              = 1n << 22n;
+    static DeafenMembers            = 1n << 23n;
+    static MoveMembers              = 1n << 24n;
+    static UseVAD                   = 1n << 25n;
+    static ChangeNickname           = 1n << 26n;
+    static ManageNicknames          = 1n << 27n;
+    static ManageRoles              = 1n << 28n;
+    static ManageWebhooks           = 1n << 29n;
+    static ManageEmojisAndStickers  = 1n << 30n;
+    static UseApplicationCommands   = 1n << 31n;
+    static RequestToSpeak           = 1n << 32n;
+    static ManageEvents             = 1n << 33n;
+    static ManageThreads            = 1n << 34n;
+    static CreatePublicThreads      = 1n << 35n;
+    static CreatePrivateThreads     = 1n << 36n;
+    static UseExternalStickers      = 1n << 37n;
+    static SendMessagesInThreads    = 1n << 38n;
+    static UseEmbeddedActivities    = 1n << 39n;
+    static ModerateMembers          = 1n << 40n;
+
+    value: bigint;
+
+    constructor(...data: PermissionResolvable) {
+        this.value = Permissions.resolve(...data);
+    }
+
+    /**
+     * @returns All the Discord permissions as a single bit.
+     */
+    static get all(): bigint {
+        return Flags.reduce((a, b) => a | b[1], 0n);
+    }
+
+    /**
+     * @returns An object containing all Discord permissions.
+     */
+    static entries(): { [key: string]: bigint } {
+        return Flags.reduce<{ [key: string]: bigint }>((a, b) => {
+            a[b[0]] = b[1];
+            return a;
+        }, {});
+    }
+
+    /**
+     * A static method for resolving permissions from strings, numbers and bigints.
+     * @param data The data to resolve permissions from.
+     * @returns The resolved bits.
+     */
+    static resolve(...data: PermissionResolvable): bigint {
+        let result = 0n;
+        const entries = this.entries();
+        for (let i of data) {
+            switch (typeof i) {
+                case 'string': if (i in entries) result |= entries[i]; break;
+                case 'number': result |= BigInt(i); break;
+                default: result |= i; break;
+            }
+        }
+        return result;
+    }
```

Comment:
I think this method should be a little more strict about what it affects; currently it simply ignores invalid input, which seems undesirable.